### ISSUE TITLE
Succeed to create snapshots for one archived emulator blob

### ISSUE THREAD

**Issue by v-xianya** [microsoft/AzureStorageExplorer#5029]
**Storage Explorer Version**: 1.22.0-dev
**Build Number**: 20211008.1
**Branch**: main
**Platform/OS**: Windows 10
**Architecture**: ia32
**Emulator Version**: Azurite 3.14.0
**How Found**: From running test case
**Regression From**: Not a regression

## Steps to Reproduce ##
1. Install and run Azurite 3.14.0 -> Launch Storage Explorer.
2. Expand 'Local & Attached -> Storage Accounts -> (Emulator - Default Ports)(Key)' -> Blob Containers.
3. Create a blob container -> Upload one blob -> Update the access tier to 'Archive'.
4. Create snapshot for the archive blob.
5. Check there is an error indicating 'This operation is not permitted on an archived blob'.

## Expected Experience ##
There is an error indicating 'This operation is not permitted on an archived blob'.

## Actual Experience ##
Succeed to create snapshot for the archived blob with no error.
![image](https://user-images.githubusercontent.com/41351993/136512653-48cb7c15-eec3-4c91-812a-0317ccbc2c46.png)

**Comment by MRayermannMSFT** [comment]
@v-xianya I had to run Azurite in `--loose` mode to repro this. Was that the case for you as well? If not, can you tell me exactly how you started Azurite?

**Comment by v-xianya** [comment]
@MRayermannMSFT 

Yes, I stated Azurite by running Azurite --loose. (This issue doesn't reproduce when starting Azurite by running Azurite)
